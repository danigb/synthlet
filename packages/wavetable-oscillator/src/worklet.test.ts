import { getProcessorName } from "./index";
import { createWorkletTestContext } from "./test-utils";

describe("WavetableOscillatorWorkletNode", () => {
  let WavetableOscillatorWorklet: any;
  const sampleRate = 40;

  beforeAll(async () => {
    createWorkletTestContext(sampleRate);
    WavetableOscillatorWorklet = (await import("./worklet"))
      .WavetableOscillatorWorklet;
  });

  it("registers processor", () => {
    expect(global.registerProcessor).toHaveBeenCalledWith(
      getProcessorName(),
      WavetableOscillatorWorklet
    );
  });

  it("has parameter descriptors", () => {
    expect(WavetableOscillatorWorklet.parameterDescriptors).toMatchSnapshot();
  });
});
