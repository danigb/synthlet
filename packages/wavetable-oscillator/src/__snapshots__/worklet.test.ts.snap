// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WavetableOscillatorWorkletNode has parameter descriptors 1`] = `
[
  {
    "automationRate": "k-rate",
    "defaultValue": 220,
    "maxValue": 20000,
    "minValue": 0,
    "name": "baseFrequency",
  },
  {
    "automationRate": "k-rate",
    "defaultValue": 440,
    "maxValue": 20000,
    "minValue": 0,
    "name": "frequency",
  },
  {
    "automationRate": "k-rate",
    "defaultValue": 0.05,
    "maxValue": 10,
    "minValue": 0,
    "name": "morphFrequency",
  },
]
`;
